erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -96,6 +96,11 @@ def on_submit(self):
 		self.update_quality_inspection()
 		if self.work_order and self.purpose == "Manufacture":
 			self.update_so_in_serial_number()
+		
+		if self.purpose == 'Send to Warehouse' and self.in_transit:
```

Comment:
Let's remove the purpose "Send to Warehouse" and "Receive at Warehouse" from stock entry. On selection of purpose as 'Material Transfer', the checkbox 'Add to Transit' will be visible and it will be checked by default. And target warehouse will be set based on the default transit warehouse.

We need to write a patch to update existing records.